craft_application/launchpad/models/recipe.py

```diff
@@ -326,11 +326,16 @@ def build(
         return self._build(deadline, request_build_kwargs)
 
 
-class CharmRecipe(_StoreRecipe):
-    """A recipe for a charm.
+class _StandardRecipe(_StoreRecipe):
```

Comment:
Given this restructuring, let's make a `SnapRecipeType` and `StandardRecipeType` up above and make those the `_resource_types` of `SnapRecipe` and `_StardardRecipe`

Reply:
as discussed, let's postpone this bit of refactoring until we have a clearer picture of this